src/reporting/reportingToken.se

```diff
@@ -98,24 +115,39 @@ def redeemWinningTokens(reporter: address):
     reporterReputationShare = reputationSupply * attotokens / self.supply
     self.supply -= attotokens
     self.tokens[reporter] = 0
-    self.getReputationToken().transfer(reporter, reporterReputationShare)
-    log(type=Transfer, reporter, 0, attotokens)
+    if (reporterReputationShare > 0):
+        self.getReputationToken().transfer(reporter, reporterReputationShare)
+        log(type=Transfer, reporter, 0, attotokens)
     return 1
 
 # FIXME: bond holders have already been paid at this point (they are paid as soon as market finalization occurs)
 # CONSIDER: track how much bond holders are owed, then as REP is extracted from losing tokens it initially goes into
 def extractBondHolderPayments():
     require(self.market.isFinalized())
     marketReputationBalance = self.getReputationToken().balanceOf(self.market)
-    desiredBondPot = self.market.getDesiredDisputeRewardPot()
+    desiredBondPot = self.getDesiredDisputeRewardPot()
     bondPotIncrease = desiredBondPot - marketReputationBalance
     if (bondPotIncrease <= 0):
-        return(0)
+        return(1)
     excessAttorepAvailable = self.getReputationToken().balanceOf(self) - self.supply
     amountToSend = min(bondPotIncrease, excessAttorepAvailable)
-    self.getReputationToken().transfer(self.market, amountToSend)
+    if (amountToSend):
+        self.getReputationToken().transfer(self.market, amountToSend)
     return 1
 
+def getDesiredDisputeRewardPot():
+    assertNoValue()
+    require(self.market.isFinalized())
+    desiredBondPot = 0
+    if (self.market.getAutomatedReporterDisputeBondToken() and self.market.getAutomatedReporterDisputeBondToken().getDisputedPayoutDistributionHash() != self.market.getFinalPayoutDistributionHash()):
+        desiredBondPot += AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT * 2
+    if (self.market.getLimitedReportersDisputeBondToken() and self.market.getLimitedReportersDisputeBondToken().getDisputedPayoutDistributionHash() != self.market.getFinalPayoutDistributionHash()):
+        desiredBondPot += LIMITED_REPORTERS_DISPUTE_BOND_AMOUNT * 2
+    if (self.market.getAllReportersDisputeBondToken() and self.market.getAllReportersDisputeBondToken().getDisputedPayoutDistributionHash() != self.market.getFinalPayoutDistributionHash()):
+        desiredBondPot += ALL_REPORTERS_DISPUTE_BOND_AMOUNT * 2
+    return(desiredBondPot)
```

Comment:
It feels like this method should be on `market` rather than on the reporting token.  I don't see any variables being accessed from this contract, they are all off of `market`.  Also, aren't those constants only defined over in `market.se` meaning this currently doesn't do the right thing?